src/language/interpreter.rs

```diff
@@ -121,25 +162,28 @@ where
                     .collect();
             slice_info[axis] = ndarray::SliceOrIndex::from(low..high);
             let slice_info = ndarray::SliceInfo::new(slice_info).unwrap();
-            access.tensor = access
-                .tensor
-                .into_owned()
-                .slice(slice_info.as_ref())
-                .into_owned();
 
-            Value::Access(access)
+            Value::Access(Access {
+                tensor: access
+                    .tensor
+                    .clone()
+                    .into_owned()
+                    .slice(slice_info.as_ref())
+                    .into_owned(),
```

Comment:
Interesting. Do you know that this exact order is necessary? I'm surprised that you need a clone and two into_owned()s, but I haven't read the docs for into_owned closely enough to fully understand what's going on here.

Reply:
Ah, I just realized you just added a `clone` to the code that was already there. Yeah, that original code is ugly and I doubt it's right...I was just trying to get it to type check 😂